js/main.js

```diff
@@ -1,2 +1,232 @@
+//this line prevents my editor from marking $ as undefined
+/* global $ */
+
 $(function () {
+  // select an option at random for the both
+  function chooseBotWeapon () {
+    const weapons = ['rock', 'paper', 'scissors']
+    const n = Math.floor(Math.random() * 3)
+    const botWeapon = weapons[n]
+    return botWeapon
+  }
+
+  // determine who wins the current hand
+  function whoWins (playerChoice, winnerList) {
+    // select the bot weapon
+    const botChoice = chooseBotWeapon()
+
+    // update the page text with current turn stats
+    $('#playerChoice').text(`${playerChoice}`)
+    $('#botChoice').text(`${botChoice}`)
+    // if the same weapons were chosen, update page to show that it was a tie.
+    // also return nobody to the winnerList since no one won the turn
+    if (playerChoice === botChoice) {
+      $('#showWinner').text('It was a tie!')
+      return 'nobody'
+    }
+
+    // logic to determine who won the current hand
+    // return the winner
+    switch (botChoice) {
+      case 'rock':
+        if (playerChoice === 'scissors') {
+          return 'bot'
+        } else if (playerChoice === 'paper') {
+          return 'player'
+        }
+        break
+
+      case 'paper':
+        if (playerChoice === 'rock') {
+          return 'bot'
+        } else if (playerChoice === 'scissors') {
+          return 'player'
+        }
+        break
+
+      case 'scissors':
+        if (playerChoice === 'paper') {
+          return 'bot'
+        } else if (playerChoice === 'rock') {
+          return 'player'
+        }
+        break
+      default:
+        console.log('Both players must choose an item!')
+        return 'nobody'
+    }
+  }
+
+  // update scoreboard page elements
+  function updateScore (winner) {
+    console.log(`running score count. adding points for ${winner}`)
+    // clear any bonus info from Scoreboard
+    $('#bonus').text('')
+    // get details from scoreboard
+    let currentHumanScore = parseInt($('#humanScore').text())
+    let currentBotScore = parseInt($('#computerScore').text())
+    // update the scoreboard and text with winner info
+    if (winner === 'player') {
+      currentHumanScore++
+      $('#showWinner').text('You win! :D')
+      $('#humanScore').text(currentHumanScore)
+    } else if (winner === 'bot') {
+      currentBotScore++
+      $('#showWinner').text('You lose :(')
+      $('#computerScore').text(currentBotScore)
+    }
+  }
+
+  // calculate bonus points for this turn
+  function calculateBonusPoints (winnerList) {
+    const n = winnerList.length - 1
+    // grab the last 3 winners from the winnerList
+    const one = winnerList[n]
+    const two = winnerList[n - 1]
+    const three = winnerList[n - 2]
```

Comment:
@sad3xoxo interesting take on getting 3 consecutive wins. another approach could be to increment a consecutive win variable for human and computer and it resets to zero when the other player scores and does not reach 3 consecutive wins. it saves memory because the `winnerList` never reset and it stores items even after a final winner. In any case, great job adding the additional features to the game!!! 👍